https://bugs.gentoo.org/show_bug.cgi?id=236779


commit 3564e5cbe4d9c0538d6eb519904ef0befab39d75
Author: Joe Marcus Clarke <marcus@freedesktop.org>
Date:   Thu Sep 4 22:13:30 2008 -0400

    Bug 17061: Handle error return from sysconf correctly
    
    	* dbus/dbus-sysdeps-unix.c:
    	* dbus/dbus-sysdeps-util-unix.c: Cast return
    	from sysconf temporarily so we actually see
    	-1.
    
    Signed-off-by: Colin Walters <walters@verbum.org>

diff --git a/dbus/dbus-sysdeps-unix.c b/dbus/dbus-sysdeps-unix.c
index 3f963bc..24a3774 100644
--- a/dbus/dbus-sysdeps-unix.c
+++ b/dbus/dbus-sysdeps-unix.c
@@ -1493,7 +1493,11 @@ fill_user_info (DBusUserInfo       *info,
     /* retrieve maximum needed size for buf */
     buflen = sysconf (_SC_GETPW_R_SIZE_MAX);
 
-    if (buflen <= 0)
+    /* sysconf actually returns a long, but everything else expects size_t,
+     * so just recast here.
+     * https://bugs.freedesktop.org/show_bug.cgi?id=17061
+     */
+    if ((long) buflen <= 0)
       buflen = 1024;
 
     result = -1;
diff --git a/dbus/dbus-sysdeps-util-unix.c b/dbus/dbus-sysdeps-util-unix.c
index 55eb934..0343a90 100644
--- a/dbus/dbus-sysdeps-util-unix.c
+++ b/dbus/dbus-sysdeps-util-unix.c
@@ -836,7 +836,11 @@ fill_group_info (DBusGroupInfo    *info,
     /* retrieve maximum needed size for buf */
     buflen = sysconf (_SC_GETGR_R_SIZE_MAX);
 
-    if (buflen <= 0)
+    /* sysconf actually returns a long, but everything else expects size_t,
+     * so just recast here.
+     * https://bugs.freedesktop.org/show_bug.cgi?id=17061
+     */
+    if ((long) buflen <= 0)
       buflen = 1024;
 
     result = -1;
